import_video.py

```diff
@@ -59,6 +59,75 @@ def get_duration(filename):
         return None
 
 class GenericImporter(object):
+
+    def __init__(self):
+        self.DEFAULT_CAMERA_NAME = 'unnamed' # used if we can't parse a camera name from video file names
+        self.REQUIRED_MODULE_CALLBACK_FUNCTIONS = ['register_camera', 'post_video_content']
+        self.DEFAULT_FILE_REGEX = ".*/(?P<camera>\w+?)\-.*\-(?P<epoch>\d+)\.mp4"
+        self.handle_args()
+
+    def handle_args(self):
+        self.parser = argparse.ArgumentParser(
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description = textwrap.dedent(DESCRIPTION), epilog=EXAMPLES
+        )
+        # optional arguments/flags
+        self.parser.add_argument('-v', '--verbose', action='store_true', default=False, help='set logging level to debug')
+        self.parser.add_argument('-q', '--quiet', action='store_true', default=False, help='set logging level to errors only')
+        self.parser.add_argument('-c', '--csv', action='store_true', default=False, help='dump csv log file')
+        self.parser.add_argument('-p', '--port', default=None, help='the segmenter port number (default: 8080)')
+        self.parser.add_argument('-p', '--host', default=None, help='the IP address or hostname of the segmenter')
+        self.parser.add_argument('-r', '--regex', default=self.DEFAULT_FILE_REGEX, 
+                help=('regex to extract input-file meta-data. The two capture group fields are <camera> and <epoch> '
+                     'which capture the name of the camera that the video originates from and the timestamp of the start of '
+                     'the video respectively. (default: "%s")' % self.DEFAULT_FILE_REGEX))
+        self.parser.add_argument('-s', '--storage', default='.processes.shelve',
+                            help='full path to the local storage db (default: ./.processes.shelve)')
+        self.parser.add_argument('-d', '--hook_data_json', default=None,
+                            help='a json object containing extra information to be passed to the hook-module')
+        self.parser.add_argument('-f', '--hook_data_json_file', default=None,
+                            help=('full path to a file containing a json object of extra info to be passed to the hook module.'
+                            'note - the values passed in through this trump the same values passed in through the `-d` param'))
```

Comment:
No important, but command line values usually trump config files, so this seems reversed to me (I would have expected `-d` to trump `-f`.